internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -355,13 +393,50 @@ func (r *AgentServiceConfigReconciler) newAgentRoute(instance *adiiov1alpha1.Age
 		if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
 			return err
 		}
-		route.Spec = routeSpec
+		route.Spec.To = routeSpec.To
+		route.Spec.Port = routeSpec.Port
+		route.Spec.WildcardPolicy = routeSpec.WildcardPolicy
```

Comment:
My guess is that there are things that are immutable once created but we might want to mention that kind of thing in a comment to prevent someone from switching it back.

Reply:
Sure, but this doesn't seem related to the goal of the PR at all.
If this somehow relates to local auth I would expect that to be explained in the commit message/PR description.

If it doesn't apply to local auth, please change this in a separate PR containing and explanation of the change.

Reply:
Good point.

Reply:
It's because the test is checking that reconcile ends without errors. 
```
result, err := ascr.Reconcile(newAgentServiceConfigRequest(asc))
```
The issue is route.Spec.Host was unspecified in the create. When unspecified, a name is automatically generated when the route is created. The problem happens when the agentRoute is reconciled a second time. This time, in mutateFn, it attempts to clear the existing value for Host. And ascr.Reconcile shows
```
      Expected
          <*errors.errorString | 0xc000748120>: {
              s: "Route's host is empty",
          }
      to be nil
```
I'll add a comment to explain why this was changed. 

Is there a better way to handle this?

Reply:
Nope, that looks like the correct solution.
Also good catch.

I know that reconcile is the public API, but I wonder if we should unit test each of the `ensure*` functions separately?
I'll add some comments in the test code about this.